handlers/whatsapp/whatsapp.go

```diff
@@ -189,55 +333,150 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	sendPath, _ := url.Parse("/v1/messages")
 	sendURL := url.ResolveReference(sendPath).String()
 
-	// TODO: figure out sending media
+	mediaPath, _ := url.Parse("/v1/media")
+	mediaURL := url.ResolveReference(mediaPath).String()
 
 	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored)
-	parts := handlers.SplitMsg(msg.Text(), maxMsgLength)
-	for i, part := range parts {
-		payload := mtPayload{
-			To:   msg.URN().Path(),
-			Type: "text",
-		}
-		payload.Text.Body = part
 
-		jsonBody, err := json.Marshal(payload)
+	// TODO: figure out sending media
+	if len(msg.Attachments()) > 1 {
```

Comment:
Also I think fo the case of more that one attachments we send the first one as media with caption text and the other  we send them alone so with caption "", that is how we handle such case in Telegram, so we want to be consistent

Reply:
Cool, how do you then handle the status updates (read, delivered, etc...) for the latter ones?

Reply:
If a message is send as multiple parts we only save the first external ID and we'll handle the status for the first one only and ignore the others since we will not find a corresponding message with the external id we did nit save.

- So a message longer that the maxLength is send as many parts and only the first part external id is saved
- If we have many attachment we send the first with text as caption then we send the others with "" caption, also still we consider the first part only for external id and status reports

Also the upload of attachment to WA can be a different method